src/Illuminate/Support/Collection.php

```diff
@@ -217,7 +217,7 @@ public function contains($key, $operator = null, $value = null)
     {
         if (func_num_args() == 1) {
             if ($this->useAsCallable($key)) {
-                return ! is_null($this->first($key));
+                return $this->first($key, '__placeholder') != '__placeholder';
```

Comment:
Is that safe though? What guarantee is there that the actual value isn't `__placeholder`?

In other places we've used an instance of `stdClass` for these.

Reply:
BTW, we should augment the `with` helper to allow passing it a callback. Then this would be:

```php
return with(new stdClass, function ($placeholder) use ($key) {
    return $this->first($key, $placeholder) != $placeholder;
});
```

Reply:
Can we _pleeeease_ get arrow functions already???

```php
return with(new stdClass, $placeholder => $this->first($key, $placeholder) != $placeholder);
```

Why can't we have nice things in PHP? :cry:

Reply:
switched to using stdClass.

Because php ¯\_(ツ)_/¯